elfi/methods/inference/samplers.py

```diff
@@ -396,25 +407,26 @@ def set_objective(self, n_samples, thresholds=None, max_iter=10,
             else:
                 self.densratio = densratio_estimation
 
-        self.adaptive_threshold = adaptive_threshold
+        # 4: Set objective
         self.objective.update(
             dict(
                 n_samples=n_samples,
                 n_batches=self.max_parallel_batches,
                 round=rounds,
                 thresholds=thresholds))
         self._init_new_round()
+        self._update_objective()
 
     def extract_result(self):
         """Extract the result from the current state.
 
-         Returns
+        Returns
         -------
         SmcSample
 
         """
```

Comment:
note! make docs indicates there's an indentation error here but i cannot figure it out :(